Automattic/jetpack
# Add Image Compare block

Adds the Image Compare block as an experimental block, the image compare blocks allows you to place two images side-by-side (or above/below)  and use a slider to reveal different aspects of them.

Here's an example:

![image-compare](https://user-images.githubusercontent.com/45363/80538575-d0b7f480-895a-11ea-85b8-9221e536d30f.gif)

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
Internal reference: p1HpG7-8NH-p2

#### Testing instructions:

* This is a beta block, so you will need to set  in wp-config.php
    `define( 'JETPACK_BETA_BLOCKS' true );`

* Create new post . Add block to post and select before/after images
* The comparison tool should show within the editor, as it will look on front-end
* Confirm resizing editor the images resize as expected
* Confirm shows on front-end and works as expected
* Test various cases of missing images or unexpected usage

#### Proposed changelog entry for your changes:

* Adds new Image Compare block 

## Future Tracking Issues;

* What to show for AMP pages that can't load library? (See: #15653)
* Fix useResizeObserver to support older (current) WordPress versions without the Gutenberg plugin (See: #15655)

## Licensing

**NOTE:** This block uses code originally from the Knight Labs Juxtapose block, I have discussed with @pesieminski and he has cleared the license notice in the readme.md and at the top of the view.js files.

Files here:
https://github.com/Automattic/jetpack/pull/15598/files#diff-6bee5a8a6f949abff285c30df8232b6e
https://github.com/Automattic/jetpack/pull/15598/files#diff-84156f4fb84990fe9b2aba4b50eef7a5
